### PR TITLE
[hotfix] Pass user to file delete

### PR DESCRIPTION
**Pull request by samanehsan**
When deleting a file node, pass in the user so that the `TrashedFileNode.deleted_by` gets set.

### CODE REVIEW

File: website/addons/base/views.py [hunk 1]
```diff
@@ -552,7 +552,7 @@ def addon_view_or_download_file(auth, path, provider, **kwargs):
 
             # Move file to trashed file node
             if not TrashedFileNode.load(file_node._id):
-                file_node.delete()
+                file_node.delete(user=auth.user)
```
Comment (felliott) [inline website/addons/base/views.py]: `user` should not be passed here.  This is the case where we have a StoredFileNode, but can't find in it in the metadata response.  i.e. it disappeared.

Comment (samanehsan) [inline website/addons/base/views.py]: ok, that makes sense

**Comment by felliott** [comment]
Tested locally, works for me!  Thanks, @samanehsan!

**Comment by icereval** [comment]
@felliott, @samanehsan test seem to be unhappy https://travis-ci.org/CenterForOpenScience/osf.io/jobs/121767087#L3319 :-/

**Pull request merged**